types/react/index.d.ts

```diff
@@ -80,7 +80,7 @@ declare namespace React {
         | (new (props: P) => Component<any, any>);
 
     interface RefObject<T> {
-        readonly current: T | null;
+        current: T | null;
```

Comment:
I think we still need the concept of a React-managed ref and a user-managed ref. The tests already have this case: `const boolRef = useRef(false)` and it expects `boolRef.current` to be a `boolean`. But with only `RefObject` `boolRef.current` will be typed as `boolean | null`. We're basically saying here that every ref needs to account for the `null` value even though it might never have that value. 

I think for many case you do want to have a bottom value in `current`. But sometimes you also don't (or want that bottom value to be something other than `null`). 

So I start playing around with moving that ` | null` member to the actual usages i.e. refs created and managed by React will contain `T | null` but everything else will just be `T`